deeplake/core/chunk_engine.py

```diff
@@ -653,7 +653,16 @@ def check_each_sample(self, samples, verify=True):
     def _sanitize_samples(self, samples, verify=True):
         check_samples_type(samples)
         if isinstance(samples, list):
-            samples = [None if is_empty_list(sample) else sample for sample in samples]
+            samples = [
+                None
+                if is_empty_list(sample)
+                or (
+                    isinstance(sample, deeplake.core.tensor.Tensor)
+                    and sample.is_empty_tensor
+                )
+                else sample
+                for sample in samples
+            ]
```

Comment:
To be honest, it would be much easier to read if you remove list comprehension and use regular for loop instead